lib/project_types/node/forms/create.rb

```diff
@@ -8,15 +8,25 @@ class Create < ShopifyCli::Form
 
       def ask
         self.title ||= CLI::UI::Prompt.ask(ctx.message("node.forms.create.app_name"))
+        self.name = check_name
         self.type = ask_type
-        self.name = self.title.downcase.split(" ").join("_")
         res = ShopifyCli::Tasks::SelectOrgAndShop.call(ctx, organization_id: organization_id, shop_domain: shop_domain)
         self.organization_id = res[:organization_id]
         self.shop_domain = res[:shop_domain]
       end
 
       private
 
+      def check_name
+        name = title.downcase.split(" ").join("_")
```

Comment:
This method is called `check_name`, but it's doing more than just checking the name, it's also formatting it.

We _could_ rename the method, e.g. `check_and_format_name`, but the word `and` in a method name is usually a sign that a method has too many responsibilities.

So another approach might be to call a separate method from `ask`, e.g.:

```
self.name = formatted_name
check_name
self.type = ask_type
```

Reply:
I see you've renamed the method but it's still doing two things. I had something like this in mind:

```ruby
def ask
  self.title ||= CLI::UI::Prompt.ask(ctx.message("node.forms.create.app_name"))
  self.name = formatted_name
  check_for_valid_name
  self.type = ask_type
  res = ShopifyCli::Tasks::SelectOrgAndShop.call(ctx, organization_id: organization_id, shop_domain: shop_domain)
  self.organization_id = res[:organization_id]
  self.shop_domain = res[:shop_domain]
end

def formatted_name
  title.downcase.split(" ").join("_")
end

def check_for_valid_name
  if name.include?("shopify")
    ctx.abort(ctx.message("node.forms.create.error.invalid_app_name"))
  end
end
```

Reply:
The current solution follows a similar format to `ask_type` and `ask_db` (in Rails); all three check for the validity of the user's input in addition to asking/formatting. So, I think `format_name` would be consistent with the existing methods.